### PR TITLE
feat: unit tests

### LINKED ISSUE #10: snap unit tests

### PR DESCRIPTION
**Pull request by BeroBurny**
Closes #10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,7 @@
 
 Metamask Snap to enable Metamask users to interact with Near dapps.
 
+## Known issue's
 
+* currently `getKeyPair` is implement using secp256k curve after metamask implement ed25519 entropy we will rewrite method to fix creating keyring with ed25519
```
Comment (mpetrunic) [inline README.md]:
```suggestion
* currently `getKeyPair` is using BIP32 with the secp256k curve. Once Metamask implements the method to get BIP32 entropy with the ed25519 curve, we will implement a fix to derive the proper key. 
```

File: packages/snap/package.json [hunk 2]
```diff
@@ -58,13 +58,17 @@
...
     "@metamask/snaps-cli": "^0.18.1",
+    "@types/bn.js": "^5",
     "@types/chai": "^4.2.10",
+    "@types/chai-as-promised": "^7.1.5",
     "@types/mocha": "^7.0.2",
     "@types/sinon": "^7.5.2",
     "@types/sinon-chai": "^3.2.3",
     "@typescript-eslint/eslint-plugin": "^5.27.1",
     "@typescript-eslint/parser": "^5.27.1",
+    "bn.js": "^5.2.1",
```
Comment (mpetrunic) [inline packages/snap/package.json:69]: why do you need bn.js?

File: packages/snap/src/rpc/signTransactions.ts [hunk 3]
```diff
@@ -22,8 +23,30 @@ export async function signTransactions(
...
+  });
+
+  // fetch data from chain - the latest block hash and latest nonce of account
+  const accessKey = await provider.query(
+    `access_key/${Buffer.from(keyPair.getPublicKey().data).toString(
+      "hex"
+    )}/${keyPair.getPublicKey().toString()}`,
+    ""
+  );
+  let nonce = ++(accessKey as unknown as { nonce: number }).nonce;
```
Comment (mpetrunic) [inline packages/snap/src/rpc/signTransactions.ts]: caller does not supply nonce?

File: packages/snap/test/unit/rpc/getAccount.test.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+chai.use(sinonChai);
+
+describe("Test rpc handler function: getAccount", function () {
+  const walletStub = mockSnapProvider();
+
+  afterEach(function () {
+    walletStub.reset();
+  });
+
+  it("should return valid address for mainnet", async function () {
```
Comment (mpetrunic) [inline packages/snap/test/unit/rpc/getAccount.test.ts:16]: test case for testnet as well

File: packages/snap/test/unit/rpc/signTransactions.test.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        network: "testnet",
+        transactions: [
+          {
+            receiverId:
+              "561ddb98e0b17cd42bf3f65b0d5147f7abff7f0d341c08cb89d31de8a788f948",
+            actions,
+          },
+        ],
+      })
+    ).to.rejectedWith();
```
Comment (mpetrunic) [inline packages/snap/test/unit/rpc/signTransactions.test.ts]: verify correct error message

File: packages/snap/test/unit/rpc/signTransactions.test.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+          actions,
+        },
+      ],
+    });
+
+    expect(result).to.not.be.null;
+    expect(result).to.be.an("array");
+  });
+
+  it("should fail on wrong action", async function () {
```
Comment (mpetrunic) [inline packages/snap/test/unit/rpc/signTransactions.test.ts:49]: add case for failed validation

File: packages/snap/package.json (hunk 2 above)
Comment (BeroBurny) [reply packages/snap/package.json:69]: `near-api-js` still uses `bn.js` for big numbers
https://github.com/near/near-api-js/blob/master/packages/near-api-js/src/transaction.ts

File: packages/snap/src/rpc/signTransactions.ts (hunk 3 above)
Comment (BeroBurny) [reply packages/snap/src/rpc/signTransactions.ts]: by the sow caller only send `receiverId`  and list of `actions` if we are going to change that caller send this data then we need to add filed `nonce` and `recentBlockHash`

File: packages/snap/package.json (hunk 2 above)
Comment (mpetrunic) [reply packages/snap/package.json:69]: but where are you using it in code?

File: packages/snap/src/rpc/signTransactions.ts (hunk 3 above)
Comment (mpetrunic) [reply packages/snap/src/rpc/signTransactions.ts]: what does near wallet selector send?

Comment (BeroBurny) [reply packages/snap/src/rpc/signTransactions.ts]:
```ts
export interface SignAndSendTransactionParams {
  signerId?: string;
  receiverId?: string;
  actions: Array<Action>;
}
```
but we can use `provider.query` to get the latest `nonce` and `recentBlockHash` (from signAndSendTransaction) from wallet-selector and make snap more as Keystore and singing tool

Comment (mpetrunic) [reply packages/snap/src/rpc/signTransactions.ts]: Let's allow optional nonce and recentBlockHash params to be sent. And if they aren't, than use provider.

Also can you create sign and signAndSend methods so we can decide later on in near-wallet-selector what is better.

FYI check note and document how are we handling it: https://github.com/near/wallet-selector/blob/main/packages/core/docs/guides/custom-wallets.md#signandsendtransactions

File: packages/snap/test/unit/wallet.mock.test.ts [hunk 7]
```diff
@@ -1,10 +1,12 @@
...
+import { SnapProvider } from "@metamask/snap-types";
 
 export class WalletMock {
   public readonly requestStub = sinon.stub();
 
   public readonly rpcStubs = {
     snap_confirm: sinon.stub(),
+    snap_getBip44Entropy_1: sinon.stub(),
     snap_getBip44Entropy_397: sinon.stub(),
     snap_manageState: sinon.stub(),
```
Comment (BeroBurny) [inline packages/snap/test/unit/wallet.mock.test.ts]: is is not used anymore and can be removed

**Review by BeroBurny — COMMENTED** [review]
@irubido found one issue

**Review by irubido — APPROVED** [review]

**Pull request merged**